src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyGroupRequest.java

```diff
@@ -0,0 +1,503 @@
+
+    private long initialId;
+    private long replyId;
+    private long routeId;
+
+    private MessageConsumer connectInitial;
+    private long connectReplyId;
+    private long connectInitialId;
+
+    private Future<?> retryRequest;
+    private int attempts;
+
+    private int state;
+
+    HttpCacheProxyGroupRequest(
+        HttpCacheProxyFactory factory,
+        HttpProxyCacheableRequestGroup requestGroup,
+        MessageConsumer initial)
+    {
+        this.factory = factory;
+        this.requestGroup = requestGroup;
+        this.initial = initial;
+        this.requestSlot =  new MutableInteger(NO_SLOT);
+        this.writeBuffer = new UnsafeBuffer(allocateDirect(factory.writer.writerCapacity()).order(nativeOrder()));
+        this.httpTypeId = factory.supplyTypeId.applyAsInt("http");
+    }
+
+    MessageConsumer newResponse(
+        HttpBeginExFW beginEx)
+    {
+        MessageConsumer newStream = null;
+        ArrayFW<HttpHeaderFW> responseHeaders = beginEx.headers();
+        boolean retry = HttpHeadersUtil.retry(responseHeaders);
+
+        if ((retry && attempts < 3) ||
+            (factory.defaultCache.checkToRetry(getRequestHeaders(),
+                                               responseHeaders,
+                                               requestGroup.getEtag(),
+                                               requestGroup.getRequestHash())))
+        {
+            final HttpCacheProxyRetryResponse cacheProxyRetryResponse =
+                new HttpCacheProxyRetryResponse(factory,
+                                                requestGroup.getRequestHash(),
+                                                connectInitial,
+                                                routeId,
+                                                connectReplyId,
+                                                this::scheduleRequest);
+            newStream = cacheProxyRetryResponse::onResponseMessage;
+        }
+        else if (isCacheableResponse(responseHeaders))
+        {
+            final HttpCacheProxyCacheableResponse cacheableResponse =
+                new HttpCacheProxyCacheableResponse(factory,
+                                                    requestGroup,
+                                                    requestSlot,
+                                                    connectInitial,
+                                                    connectReplyId,
+                                                    routeId,
+                                                    this::scheduleRequest);
+            newStream = cacheableResponse::onResponseMessage;
+        }
+        else
+        {
+            Function<HttpBeginExFW, MessageConsumer> responseFactory = factory.correlations.remove(replyId);
+            if (responseFactory != null)
+            {
+                MessageConsumer newResponse = responseFactory.apply(beginEx);
+                newStream = (t, b, i, l) ->
```

Comment:
Please move this lambda to a method, say `onResponseMessage`.

Reply:
I believe you are saying that I should move `(t, b, i, l) ->` to a method, however, it needs access to `newResponse`. Please correct me if I am wrong.